listenerutil/parse.go

```diff
@@ -377,15 +382,15 @@ func ParseListeners(list *ast.ObjectList) ([]*ListenerConfig, error) {
 		// HTTP Headers
 		{
 			// if CustomApiResponseHeadersRaw is nil, we still need to set the default headers
-			customApiHeadersMap, err := parseCustomResponseHeaders(l.CustomApiResponseHeadersRaw, false)
+			customApiHeadersMap, err := parseCustomResponseHeaders(l.CustomApiResponseHeadersRaw, false, WithDefaultResponseHeaders(opts.withDefaultApiResponseHeaders))
 			if err != nil {
 				return nil, multierror.Prefix(fmt.Errorf("failed to parse custom_api_response_headers: %w", err), fmt.Sprintf("listeners.%d", i))
 			}
 			l.CustomApiResponseHeaders = customApiHeadersMap
 			l.CustomApiResponseHeadersRaw = nil
 
 			// if CustomUiResponseHeadersRaw is nil, we still need to set the default headers
-			customUiHeadersMap, err := parseCustomResponseHeaders(l.CustomUiResponseHeadersRaw, true)
+			customUiHeadersMap, err := parseCustomResponseHeaders(l.CustomUiResponseHeadersRaw, true, WithDefaultResponseHeaders(opts.withDefaultUiResponseHeaders))
```

Comment:
Using this option as implemented will overwrite both UI and API response headers - should we not ensure this option only overwrites the UI response headers?